### PR TITLE
add support for hexagon-unknown-linux-musl

### PR DESCRIPTION
**Pull request by androm3da**

### CODE REVIEW

File: src/librustc_codegen_llvm/llvm_util.rs [hunk 1]
```diff
@@ -174,7 +174,7 @@ const X86_WHITELIST: &[(&str, Option<Symbol>)] = &[
 
 const HEXAGON_WHITELIST: &[(&str, Option<Symbol>)] = &[
     ("hvx", Some(sym::hexagon_target_feature)),
-    ("hvx-double", Some(sym::hexagon_target_feature)),
+    ("hvx-length128b", Some(sym::hexagon_target_feature)),
```
Comment (androm3da) [inline src/librustc_codegen_llvm/llvm_util.rs]: This isn't strictly part of the `hexagon-unknown-linux-musl` target support, just found this mismatch while reviewing.  I can leave this part out of the change if that makes more sense.

File: src/libunwind/lib.rs [hunk 2]
```diff
@@ -22,7 +22,11 @@ cfg_if::cfg_if! {
...
 }
 
-#[cfg(target_env = "musl")]
+#[cfg(all(not(target_arch = "hexagon"), target_env = "musl"))]
 #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
 #[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
 extern {}
+
+#[cfg(all(target_arch = "hexagon", target_env = "musl"))]
+#[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
```
Comment (alexcrichton) [inline src/libunwind/lib.rs]: This seems slightly suspicious in the sense that the above should work even for the hexagon target I think? Is it possible to remove this change, and if not what does the error look like?

Comment (androm3da) [reply src/libunwind/lib.rs]: Sorry, I think this is a stale change that was merged from an older baseline where it made some sense.

I'll remove it.

**Pull request merged**